### PR TITLE
Revert "feat: fetching profiles from server"

### PR DESCRIPTION
**Pull request by hansac91**
Reverts nerdbord/good-dev-hunting-app#51

### CODE REVIEW

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
# Code Review Report

## Task Goal
Fetch profiles list from Next.js API endpoint using Next.js server components, so profiles are fetched server side, not client side.

## Code Changes Review

### app/page.tsx
The change in this file is removing the comment `// @ts-expect-error Server Component`. This change doesn't affect the functionality of the code.

### src/components/ProfileList/ProfileList.tsx
The changes in this file are extensive. The `ProfileList` component has been completely rewritten. The component now uses client-side data (`profileData`) instead of fetching data from the server. This is a clear violation of the task requirements which state that data should be fetched from a Next.js API endpoint using server components.

The `ProfileListItem` component has been moved into this file from its own file. This component now also uses client-side data (`data`) instead of data fetched from the server.

### src/components/ProfileList/ProfileListItem.tsx
This file has been removed. The `ProfileListItem` component that was in this file has been moved to `src/components/ProfileList/ProfileList.tsx`.

### src/components/ProfileList/ProfilesWithFilter.tsx
This file has been removed. The functionality that was in this file has been moved to `src/components/ProfileList/ProfileList.tsx`.

### src/components/ProfileList/filters.ts
The changes in this file are renaming and updating the filter functions to work with the new client-side data structure. These changes are consistent with the changes in `src/components/ProfileList/ProfileList.tsx`.

## Conclusion
The code changes do not meet the task requirements. The task requires fetching data from a Next.js API endpoint using server components, but the code changes use client-side data instead. The code needs to be rewritten to fetch data from the server as per the task requirements.

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
# Code Review Report

## Task goal
Fetch profiles list from Next.js API endpoint

## Requirements
Fetch should occur using [Next.js server components](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching), so profiles are fetched server side, not client side

## Resource endpoint
```bash
curl -X GET \
  http://localhost:3000/api/profiles
```

## Review

The task requirement was to fetch profiles list from a Next.js API endpoint using server components. However, the provided code changes do not seem to fulfill this requirement. 

In the modified file `src/components/ProfileList/ProfileList.tsx`, the data is being fetched from a local file `profile-data` instead of the API endpoint. The `getPublishedProfilesPayload` function that was previously used to fetch data from the backend has been removed, and no new code has been added to fetch data from the API endpoint.

The `ProfileListItem` component has been moved from `ProfileListItem.tsx` to `ProfileList.tsx`, and the `ProfilesWithFilter.tsx` file has been removed. These changes do not seem to be related to the task requirement.

The changes in `filters.ts` are just updates to the existing filter functions to work with the new data structure. These changes are also not related to the task requirement.

In conclusion, the provided code changes do not cover the task requirements. The profiles are not being fetched from the Next.js API endpoint using server components. Instead, they are being fetched from a local file.
